### PR TITLE
C#6-ifying all the things

### PR DESCRIPTION
**Pull request by mwhelan**
A pass through Seleno project to consistently and (hopefully) comprehensively apply new C#6 features.

### CODE REVIEW

File: src/TestStack.Seleno/Configuration/AppConfigurator.cs [hunk 1]
```diff
@@ -52,7 +51,7 @@ public ISelenoApplication CreateApplication()
 
         private IContainer BuildContainer()
         {
-            ContainerBuilder.Register(c => new ProxyGenerator(disableSignedModule: true))
+            ContainerBuilder.Register(c => new ProxyGenerator(true))
```
Comment (robdmoore) [inline src/TestStack.Seleno/Configuration/AppConfigurator.cs]: I prefer the label here - passing in a random boolean isn't very intention revealing

**Comment by robdmoore** [comment]
Rest looks good.

Is it worth mentioning this in the readme so that if people don't have VS2015 installed they understand where the problem is?

**Comment by mwhelan** [comment]
Yes, that's a good idea about the readme. I also agree about reverting the bool parameter to show the label as being more intention revealing. 

Thanks for the review, @robdmoore

**Comment by robdmoore** [comment]
Feel free to merge after changing that stuff :)

**Comment by mwhelan** [comment]
I did not add the comment to readme about C#6 and VS2015. I did some research and it seems you can setup VS2013 to work with C#6.

**Pull request merged**